### DISCUSSION TITLE
Is there a method to set a local storage value?

### DISCUSSION THREAD

**Discussion by STARKILLER-1** [seleniumbase/SeleniumBase#2722] [Q&A] [answered]
Is there a method to set a local storage value? I am doing it by `driver.execute_script(f"window.localStorage.setItem('wbx__tokenData', '{json.dumps({'token': token})}')")` but when I change page local storage clears my token

**Comment by mdmintz** [comment, marked as answer]
For the `SB()` or `BaseCase` formats, you can use `set_local_storage_item(key, value)` to set a `local_storage` value.

https://github.com/seleniumbase/SeleniumBase/blob/b77a1c022bb4ebf8cc1fec4db92847a14f9bc818/help_docs/method_summary.md?plain=1#L602

**Reply by STARKILLER-1** [reply]
And for the Driver() object?

**Reply by mdmintz** [reply]
The `Driver()` is limited compared to `SB()`. Either use `SB()` if you need the extra methods/functionality, or create your own methods that do what you need.

**Reply by leesache** [reply]
Hello, Michael. Why does local storage variable, which i've set via set_local_storage_item, after refreshing page is deleted?

```
            with SB(
                    uc=True,
                    browser="chrome",
                    locale="ru",
                    position="0,0",
                    agent=browser_config.get("user_agent"),
                    window_size=browser_config.get("window_size"),
                    ad_block=True,
                    devtools=True
            ) as driver:

```

```
    def login(driver: SeleniumBase, token: str):
    logger.info("Execute token to LocalStorage")
    temp = json.dumps({"token": token})
    driver.set_local_storage_item("wbx__tokenData", temp)
```


While we were working with raw selenium we did it like this:

```
    driver.execute_script("localStorage.setItem(arguments[0], JSON.stringify({token: arguments[1]}));",
                      'wbx__tokenData', token)
```
And it were working fine.

**Reply by mdmintz** [reply]
As of `4.35.7` you have these CDP Mode methods:

```python
sb.cdp.set_local_storage_item(key, value)
sb.cdp.get_local_storage_item(key)
```

And if you're calling things from regular mode, use:

```python
sb.set_local_storage_item(key, value)
sb.get_local_storage_item(key)
```

Any method called from `driver` is using the raw Selenium API... (not the SeleniumBase one).
If there's an issue with a raw Selenium method, then you would have to check with their repo.
